WlngraphConvolution and WLNScoringModel

Sauravroy34 opened the pull request:
## Description

Fix #(issue)

<!-- Please include a summary of the change and which issue is fixed.
Please also include relevant motivation and context.
List any dependencies that are required for this change. -->


## Type of change
#2680


Added wln graph convolution and scoring model 
as per 
1) [Predicting Organic Reaction Outcomes with
Weisfeiler-Lehman Network](https://arxiv.org/pdf/1709.04555)
2) [A graph-convolutional neural network model for the prediction of chemical reactivity](https://pubs.rsc.org/en/content/articlelanding/2019/sc/c8sc04228d)
3) [rexgen_direct](https://github.com/connorcoley/rexgen_direct)
Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

JoseAntonioSiguenza commented on deepchem/models/torch_models/WLN.py:
```diff
@@ -0,0 +1,115 @@
+import torch
```
please make file name lowercase (`wln.py`)

Sauravroy34 replied:
done

JoseAntonioSiguenza commented on deepchem/models/torch_models/WLN.py:
```diff
@@ -0,0 +1,115 @@
+        self.hidden_size = hidden_size
+        self.num_bond_orders = num_bond_orders
+
+    def forward(self, atom_features, adj_matrix, wln_bond_features, binary_features, atom_mask):
```
please add type annotations

JoseAntonioSiguenza commented on deepchem/models/torch_models/WLN.py:
```diff
@@ -0,0 +1,115 @@
+        return scores
+    
+
+class WLNScoring(TorchModel):
```
Please add dosctrings, usage examples, etc.

JoseAntonioSiguenza commented on deepchem/feat/molecule_featurizers/wln_featurizer.py:
```diff
@@ -0,0 +1,513 @@
+import numpy as np
```
Please add tests

Sauravroy34 commented:
@JoseAntonioSiguenza 
Thanks for the initial review!
This is the pr for the model
i have added unit tests (for the model) and added the model to docs as well
would love to get a second reveiw on this pr!

#4585 This is the pr for the featurizer ( i have edited it to match MolGAN featurizer style) 
changed the class Name to be more descriptive 
and for now adding unit tests and adding the featurizer to docs is remaining
